src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotRemoveBaseTypeOrInterface.cs

```diff
@@ -38,24 +38,25 @@ private void RunOnTypeSymbol(ITypeSymbol left, ITypeSymbol right, string leftNam
 
         private void ValidateBaseTypeNotRemoved(ITypeSymbol left, ITypeSymbol right, string leftName, string rightName, IList<CompatDifference> differences)
         {
-            // Base type order matters in the hierarchy chain so therefore use a list.
-            List<ITypeSymbol> rightBaseTypes = right.GetAllBaseTypes().ToList();
+            ITypeSymbol leftBaseType = left.BaseType;
+            ITypeSymbol rightBaseType = right.BaseType;
 
-            int lastTypeIndex = 0;
-            foreach (ITypeSymbol leftBaseType in left.GetAllBaseTypes())
+            while (rightBaseType != null)
             {
-                lastTypeIndex = rightBaseTypes.FindIndex(lastTypeIndex, r => Settings.SymbolComparer.Equals(leftBaseType, r));
-
-                if (lastTypeIndex < 0)
-                {
-                    differences.Add(new CompatDifference(
-                                            DiagnosticIds.CannotRemoveBaseType,
-                                            string.Format(Resources.CannotRemoveBaseType, left.ToDisplayString(), leftBaseType.ToDisplayString(), rightName, leftName),
-                                            DifferenceType.Changed,
-                                            right));
+                // If we found the immediate left base type on right we can assume
+                // that any removal of a base type up on the hierarchy will be handled
+                // when validating the type which it's base type was actually removed.
+                if (Settings.SymbolComparer.Equals(leftBaseType, rightBaseType))
                     return;
-                }
+
+                rightBaseType = rightBaseType.BaseType;
```

Comment:
Should strict mode break and fallthrough to difference diagnostic?

Reply:
Ah that's a good point. I think StrictMode in this case would let right to drop a base type that left has and not catch it. I think StrictMode should match and should just look for `left.BaseType == right.BaseType` rather than searching the tree for the presence of right on left.

Reply:
Hmm actually I tried this and didn't like the outcome of the diagnostics, because if one side drops the basetype, i.e right drops it, then the message would echo saying that left doesn't inherit from object but right does. 

I think it is better to try and search the tree and if right adds a new base type it would be handled correctly the way it is.